lib/p2p/Peer.ts

```diff
@@ -122,10 +131,12 @@ class Peer extends EventEmitter {
     await this.initHello();
     this.finalizeOpen();
 
-    const handshakeState: HandshakeState = {};
-
     // let the pool know that this peer is ready to go
-    this.emit('open', handshakeState);
+    this.emit('open', this.handshakeState!);
+
+    // request peer's orders and known hosts
+    this.sendPacket(new packets.GetOrdersPacket());
+    this.sendPacket(new packets.GetHostsPacket());
```

Comment:
putting this here is a bit misleading, because it's post-handshake. why not in `Pool.handleOpen`? That's where these stuff should be.

Reply:
I'll move it back.